### PR TITLE
pAIs are now used in MODsuits instead of AIs

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Replaces AIs being inserted into MODsuits with pAIs.

I know I worked on this during the feature freeze, but I've done it because I was working on https://github.com/tgstation/tgstation/pull/63688 and was very interested to see how viable it would be to make pAIs fit in MODsuits, and ended up doing 80% of the work before realizing, as I very easily get lost in whatever I do.

### The problem with AIs on MODsuits:
- AIs generally control large pieces of equipment, like Cyborgs and Mechs, with control beacons. having your intellicard inserted isn't consistent with this. Intellicards are meant to transfer you to a reconstruction console when you're dead, not put you in someone's suit
- AIs, even while in a MODsuit, can hear all radio channels. This means the person wearing the suit the AI is in can ALSO hear all channels, regardless of their job. A Roboticist can hear Command comms with this, because I've seen Robos build an AI for their modsuits nearly every round because MODsuit AIs are a cool feature
- AIs can still access door while in a MODsuit. It seems rather obvious, but your everyday Roboticist with a personal AI opening/bolting/shocking any door they want, isn't a good idea.
- If you're using the Station AI for your MODsuit, you're taking the AI away from doing anything for the main station, and taking them away from being able to command their Cyborgs to do anything, leaving them AI-less. You're also ruining the round for the AI if it rolled Malf, as it will be revealed roundstart before they had a chance to even do anything, just because you wanted to stuff them in a suit.
- AIs have binary comms. Malf AIs are already a rare antagonist to roll, now imagine having it ruined because Robotics wanted to build 3 AIs to hand them out to people's MODsuits. I already had a MODsuit AI tell me (captain) about a malf AI and I just ignored them because it's just such a scummy thing.
- AIs cannot see into Maintenance. Wearing a MODsuit with an AI in it basically forces players to stay out of maintenance if they don't want to just bore the AI inside, out.

### Why pAI's are the solution:
- Dumb argument, but it's just in the name. An AI in your MODsuit is your personal AI, and pAIs are short for Personal AI.
- You choose the personality. pAI players input their name, information, OOC notes, ect ect. You know exactly who you're downloading, even better if they write in their OOC notes "I'm fine with being a MODsuit" or something similar.
- pAIs don't care for station cameras and can see all around them, without the problem of seeing nothing but static if they get thrown down disposals.
- pAIs are still Silicon. They have laws they have to follow. Positronic brains can be inserted into mechs, but Posibrains still don't have laws, what they have to do in there is basically up to whatever policy there is in place, something I'm not a fan of (though I am likely biased as I'm from fulp and downstreams tend to have different policies)
- pAIs still have their pAI modules that they can access, rather than solely being a MODsuit. They can take pictures within the suit and send it to people over PDA/newscaster. Just being a pAI in general, unlike Positronic brains which would do nothing but sit around in a suit, not even important enough to have a name.
- pAIs can already be inserted into small bots to do small tasks. They can be put into Medibots to heal, health/chem scan, and even grab people, with Medical/Robotics access as well. Being in a MODsuit just means they can control the modsuit and move the user around, using their access by moving the user into a door.
^ Adding onto this ^ 
While being able to move the user sounds powerful, it also drains your MODsuit's cell quite a bit to do so, so while you CAN move around an unconscious user, you can't for very long. And as you're an AI, not a pAI, you can't just go recharge yourself whenever you need. Think of it as a trade-off, a pAI in a MODsuit can move around their owner, which they can't in holochassis, but not for very long, compared to holochassis' lack of a cell requirement.
- This also doesn't make pAI's independent in any way, they cannot insert themselves into a MODsuit, cannot remove themselves from one, cannot recharge themselves while within one, and rely on a conscious owner to take care of all these things that they cannot.

Now that I've explained that, I'd also like to explain
### Why Positronic brains **aren't** the solution (mostly just pAI points but reversed):
- Posibrains aren't assigned names, outside of a list of coder references, until they enter a shell, at which point they will get the name from their preferences menu.
- The player cannot choose who takes over the brain nor if the positronic brain even wants to go in a modsuit instead of a borg shell. Compare this to pAI's OOC notes section.
- Posibrains are MMIs but intended for ghost roles instead of debraining an actual player. Giving posibrains something MMI's can't would make them more than what they are intended to be.
- Like AIs, posibrains are meant to control important roles. AIs, Robots and Mechs are all things that posibrains can be put in, and are all more important roles than sitting in someone's suit.
- They don't have laws. What they have to do and who they obey is all up to Policy, something I am not a fan of. This isn't something I can easily change, either, because it would be taking away something that Admins will consider to be part of their domain as Policy, not something that should be forced in code. This is because:
- Posibrains aren't Silicons. They're just objects, subtypes of MMIs, actually. There's no silicon involved at all in the brain alone.

AI tech is required to print AI core boards in the first place, outside of the 1 board in RD's office. Law boards aren't hard to obtain either, since all circuit imprinters print them, for cheap if upgraded. If anything, having to set an AI to be a personal AI before putting it in your MODsuit sounds worse for AIs.

> It is too easy to get a pAI compared to AIs

pAIs are meant to be easier to get, so while it does make it easier to get, changing the suit from taking AIs to pAIs is also a pretty massive nerf, since they no longer get all the AI abilities that come along with it. Now it's JUST the suit being controlled, nothing more. Unlike AIs, pAIs cannot get in a suit recharger on their own, as they cannot mess with the game world, outside of moving around.

### Why was AIs put in there in the first place then?
image

### Do they WANT pAIs to be in there ANYMORE?
image

## Changelog

:cl:
balance: AIs controllers MODsuits cannot move an incapacitated or dead owner anymore.
refactor: Modsuits have been refactored to allow pAIs to enter them.
/:cl:

### CODE REVIEW

File: code/_onclick/hud/action_button.dm [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #define ACTION_BUTTON_DEFAULT_BACKGROUND "default"
+#define ACTION_BUTTON_PAI_BACKGROUND "bg_tech"
```
Comment (Fikou) [inline code/_onclick/hud/action_button.dm:2]: you dont need this, you use this define in only one place and that isnt even what actual pai backgrounds look like

File: code/modules/mod/mod_control.dm [hunk 2]
```diff
@@ -206,10 +200,10 @@
 			. += span_notice("You could remove the cell with an <b>empty hand</b>.")
 		else
 			. += span_notice("You could use a <b>cell</b> on it to install one.")
-		if(ai)
-			. += span_notice("You could remove [ai] with an <b>intellicard</b>.")
+		if(mod_pai)
+			. += span_notice("You could remove [mod_pai] with <b>Right Click</b>")
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:204]:
```suggestion
			. += span_notice("You could remove [mod_pai] with <b>right-click</b>.")
```

File: code/modules/mod/mod_control.dm [hunk 3]
```diff
@@ -206,10 +200,10 @@
...
 		else
-			. += span_notice("You could install an AI with an <b>intellicard</b>.")
+			. += span_notice("You could install a pAI with an <b>pAI card</b>.")
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:206]:
```suggestion
			. += span_notice("You could install a pAI with a <b>pAI card</b>.")
```

File: code/modules/mod/mod_control.dm [hunk 4]
```diff
@@ -206,10 +200,10 @@
 			. += span_notice("You could remove the cell with an <b>empty hand</b>.")
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:200]:
```suggestion
			. += span_notice("You could remove the cell with <b>left-click</b>.")
```

File: code/modules/mod/mod_control.dm [hunk 5]
```diff
@@ -287,6 +281,14 @@
 		return
 	return ..()
 
+/obj/item/mod/control/attack_hand_secondary(mob/user, modifiers)
+	if(seconds_electrified && cell?.charge)
+		if(shock(user))
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:286]: can put this in the previous if

File: code/modules/mod/mod_control.dm [hunk 6]
```diff
@@ -287,6 +281,14 @@
...
+			return
+	if(open && loc == user)
+		remove_pai(user)
+	return ..()
```
Comment (Fikou) [inline code/modules/mod/mod_control.dm:290]: do we want to return ..() if we remove the pai?

File: code/modules/mod/mod_pai.dm [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+	mod_pai = null
+
+#define MOVE_DELAY 2
+#define WEARER_DELAY 1
+#define LONE_DELAY 5
+#define CELL_PER_STEP DEFAULT_CELL_DRAIN * 2.5
+#define PAI_FALL_TIME 1 SECONDS
+
+/obj/item/mod/control/relaymove(mob/user, direction)
+	if((!active && wearer) || !cell || cell.charge < CELL_PER_STEP  || user != mod_pai || !COOLDOWN_FINISHED(src, cooldown_mod_move) || (wearer?.pulledby?.grab_state > GRAB_PASSIVE))
```
Comment (Fikou) [inline code/modules/mod/mod_pai.dm:52]:
```suggestion
	if((!active && wearer) || !cell || cell.charge < CELL_PER_STEP  || user != mod_pai || !COOLDOWN_FINISHED(src, cooldown_mod_move) || (wearer?.pulledby?.grab_state))
```

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/modules/mod/mod_control.dm (hunk 5 above)
Comment (JohnFulpWillard) [reply code/modules/mod/mod_control.dm:286]: this was copied over from attack_hand, do you want me to do it there too?

Comment (Fikou) [reply code/modules/mod/mod_control.dm:286]: sure

**Review by Fikou — APPROVED** [review]
lgtm

**Review by Timberpoes — CHANGES_REQUESTED** [review]
As per what I mentioned on Discord, pAIs should not be able to move modsuits for their owners.

I do not care that pAIs can currently inhabit simplebots and interact with the world. I consider that the exception, not the rule.

pAIs still have a host of useful abilities to a modsuit owner including potentially inbuilt Med and SecHUDs, universal translator, GPS and the option for them to have additional comms channels.

pAIs are primarily an RP-focused ghost role with a host of tools to support players of any job or role. I want to continue that design approach as a more indirect support role.

I have no problem with the pAI controlling the **internals** of the suit like module selection or whatever else.

I have no problem with the suit owner sharing any of the benefits of the pAI's purchased upgrades either directly or indirectly. This would make pAIs a valuable companion for a modsuit owner if they end up fully integrated into the suit while inserted.

I have no problem with an internal comms channel between the suit owner and the pAI so every modsuit owner has their own private channel to their pAI while in the suit.

I've seen comments about wanting modsuits to be like Iron Man suits with their own integrated AIs. pAIs can definitely fill this niche, except moving the suits for their owners.

Those sort of interactions like that are neat and let pAIs enhance the modsuit's capabilities without pAIs being able to move suits for dead or incapacitated owners.

If you want a modsuit to be moved externally, you can always allow an AI to jump into a modsuit with the appropriate upgrade similar to how they can jump into one of their spare shells and allow them more powerful control over it.

Things such as the pAI doorjack ability are heavily limited. They require someone to physically plug in the doorjack (in other words, they can't be dead or incapacitated and the pAI can't initiate it autonomously), requires a decently long time and alerts all AIs including the location of the hack.

Requiring the pAI to be inserted into the suit isn't a limitation any more than requiring a pAI to be within melee range of you when you want to start its door jack ability is a limitation.

**Comment by JohnFulpWillard** [comment]
> I do not care that pAIs can currently inhabit simplebots and interact with the world. I consider that the exception, not the rule.

I completely disagree with this argument. We shouldn't have mechanics around something and just tell anyone who looks at it that it's an exception and not a rule. It should either be an example of what is allowed, or it should be just removed.

> pAIs still have a host of useful abilities to a modsuit owner including potentially inbuilt Med and SecHUDs, universal translator, GPS and the option for them to have additional comms channels.

There was a suggestion to make MODsuit capabilities into a pAI purchasable thing, like Med/SecHUDs. I could implement that if needed, but as it was only suggested once and never mentioned again, I didn't think it was a popular enough idea to try.
The additional comms channels also requires the encryption keys to be manually inserted, by the master, just like how MODsuits require the pAI being inserted, and how they require the master to make the conscious effort to recharge the suit so the pAI can continue to do things.

> I have no problem with the suit owner sharing any of the benefits of the pAI's purchased upgrades either directly or indirectly. This would make pAIs a valuable companion for a modsuit owner if they end up fully integrated into the suit while inserted.

I do have a problem with it because I don't see the pAI and the master as the same person, so I don't want to implement this.

> I have no problem with an internal comms channel between the suit owner and the pAI so every modsuit owner has their own private channel to their pAI while in the suit.

I don't like this either, it sounds like it'll just be some holoparasite knockoff. Whispering exists anyhow.

> If you want a modsuit to be moved externally, you can always allow an AI to jump into a modsuit with the appropriate upgrade similar to how they can jump into one of their spare shells and allow them more powerful control over it.

This is something I don't like, which I already explained in the PR body.

> Things such as the pAI doorjack ability are heavily limited. They require someone to physically plug in the doorjack (in other words, they can't be dead or incapacitated and the pAI can't initiate it autonomously), requires a decently long time and alerts all AIs including the location of the hack.

pAIs in MODsuits are also limited by its power cell. AIs can charge themselves as AIs in modsuits can still interact with machinery/doors due to still being an AI. pAIs cannot, so require a master to consciously do it. 

> Requiring the pAI to be inserted into the suit isn't a limitation any more than requiring a pAI to be within melee range of you when you want to start its door jack ability is a limitation.

Moving around in the MODsuit drains charge relatively fast, so unlike simplemobs who can just move people around forever, MODsuits can only go so much time before they run out of charge and need to go in a recharging station, which wouldn't be possible if the host is dead.

**Comment by Nistishere** [comment]
Not a particular fan of the suit moving an incapped/corpse about myself. However, the fact that it's a pAI makes it at least more bearable, no longer having to deal with the portable AI shit means that the pAI by itself can't really go that far without doorjack, or a lot of time and patience.

**Comment by Likteer** [comment]
I would like to add, that I recall something about pAIs already being insertable into ninja suits.

**Comment by Timberpoes** [comment]
> > I do not care that pAIs can currently inhabit simplebots and interact with the world. I consider that the exception, not the rule.
> 
> I completely disagree with this argument. We shouldn't have mechanics around something and just tell anyone who looks at it that it's an exception and not a rule. It should either be an example of what is allowed, or it should be just removed.
> 

It is the exception, not the rule.

I don't care whether that mechanic gets removed or not, although I am going to close any knee-jerk removal I suspect is made on the grounds of "if I can't have what I want from pAIs, nobody else can have anything similar". The removal will have to be fully justified with reasons that go further beyond "exceptions are bad".

> > pAIs still have a host of useful abilities to a modsuit owner including potentially inbuilt Med and SecHUDs, universal translator, GPS and the option for them to have additional comms channels.
> 
> There was a suggestion to make MODsuit capabilities into a pAI purchasable thing, like Med/SecHUDs. I could implement that if needed, but as it was only suggested once and never mentioned again, I didn't think it was a popular enough idea to try. The additional comms channels also requires the encryption keys to be manually inserted, by the master, just like how MODsuits require the pAI being inserted, and how they require the master to make the conscious effort to recharge the suit so the pAI can continue to do things.

I don't mind the pAI integrating its own modules into the modsuit. I think it's a pretty cool feature and genuinely enhances the modsuit while a pAI is integrated into it.

> > I have no problem with the suit owner sharing any of the benefits of the pAI's purchased upgrades either directly or indirectly. This would make pAIs a valuable companion for a modsuit owner if they end up fully integrated into the suit while inserted.
> 
> I do have a problem with it because I don't see the pAI and the master as the same person, so I don't want to implement this.
> > I have no problem with an internal comms channel between the suit owner and the pAI so every modsuit owner has their own private channel to their pAI while in the suit.
> 
> I don't like this either, it sounds like it'll just be some holoparasite knockoff. Whispering exists anyhow.

You don't have to implement anything if you don't want to. But I am not allowing pAIs to move modsuits autonomously.

I was coming up with alternative ways to integrate pAIs into modsuits that were based around ideas other than them moving modsuits.

If your sole goal with this PR was getting pAIs to move modsuits and you're not willing to entertain other ideas for integrating pAIs into modsuits, then this isn't going to be a very productive discussion.

> > If you want a modsuit to be moved externally, you can always allow an AI to jump into a modsuit with the appropriate upgrade similar to how they can jump into one of their spare shells and allow them more powerful control over it.
> 
> This is something I don't like, which I already explained in the PR body.

The idea of inserting an IntelliCard or Posibrain or whatever is dumb, I agree there.

In my broader vision, it would involve either an AI requesting itegration or a player requesting AI integration. At that point, the modsuit would become like a pseudo shell allowing the AI to hop into and out of the suit with complete control over the modsuit's systems.

The difference with this feature is that the AI would remain in their core. They would still be subvertable (even while in the modsuit, similar to being in a shell) and the modsuit owner would not always be able to trust the AI in their modsuit because of this. The modsuit where an AI has been granted the ability to integrate would be treated as a form of AI shell in that respect.

You don't have to implement it if you don't want to. I was coming up with alternative ways to autonomously move modsuits that may be more acceptable to me and may be more acceptable to other maintainers. I can't speak for them though, so who knows.

> > Things such as the pAI doorjack ability are heavily limited. They require someone to physically plug in the doorjack (in other words, they can't be dead or incapacitated and the pAI can't initiate it autonomously), requires a decently long time and alerts all AIs including the location of the hack.
> 
> pAIs in MODsuits are also limited by its power cell. AIs can charge themselves as AIs in modsuits can still interact with machinery/doors due to still being an AI. pAIs cannot, so require a master to consciously do it.

That is fine.

However, the primary use case for pAIs in modsuits is not walking to charging stations. It's walking dead or incapacitated modsuit occupants to safer places such as medbay, a safer hallway or a room with oxygen in.

I'm going to assume the modsuit is not gonna run out of charge after 10 tiles and as a result I am not allowing pAIs to move modsuits autonomously because I do not want this use-case to happen. I do not like the concept of a pAI being able to move a modsuit when the owner is dead or incapacitated.

> > Requiring the pAI to be inserted into the suit isn't a limitation any more than requiring a pAI to be within melee range of you when you want to start its door jack ability is a limitation.
> 
> Moving around in the MODsuit drains charge relatively fast, so unlike simplemobs who can just move people around forever, MODsuits can only go so much time before they run out of charge and need to go in a recharging station, which wouldn't be possible if the host is dead. It's not the SAME limitation, it's an alternate one. MODsuit pAIs get some benefits and drawbacks, like I feel it should be.

No matter how much charge it drains, if it's enough for a pAI to forcibly move a modsuit's owner out of a dangerous situation and into a safer one then I do not want it.

---

To me, the big benefit of having a pAI in your modsuit is having a second player who is guaranteed completely loyal to you and whose sole purpose is to benefit you so that you are never alone. You can never be spirited away silently as long as the pAI can use their radio. They have various HUDs. They can translate languages. The only way they will ever be able to betray you is if they get stolen and emagged.

All of that AND being able to move you around when you're dead or incapped is just too much for a companion that you can guarantee is loyal to you at a "if they don't follow my orders I can ahelp it" level.

I love pAIs for their roleplay utility. They're big ol' gamer gear kits that are linked to another actual player. You have to both work together and communicate for their abilities to function. I have no issues with these aspects being incorporated into modsuits. But I draw the line at pAIs moving modsuits.

**Comment by EricZilla** [comment]
I agree with the change but some of the reasons are silly
The intellecard is not exclusively for rebuilding AIs, people can make pocket AI if they feel like it, even without modsuits

**Comment by Mothblocks** [comment]
you should keep ais in modsuits and let them move you around but not let pais

**Comment by Mothblocks** [comment]
also i like that with an AI i can take an existing person and put them inside, i can't do this with pAIs without metagaming

**Review by Mothblocks — CHANGES_REQUESTED** [review]
actually i'm going to promote my concerns, keep AIs in modsuits and let them move around if timber doesn't want pAIs to be able to

**Comment by zxaber** [comment]
How about a middle-ground? Allow pAIs to force a suit to crawl (as if clunkily controlling an arm servo) to let the pAI drag an incapacitated owner to safety or medical, but not allow regular walking if the user happens to be dead.

I wanted to hate the PR, but it's got a point. Normal AIs are supposed to be managing the station, or at least acting mostly independently to fufill their laws. A pAI is intended be tagging along their human(oid) owner and assisting directly with basic tasks, and it makes a lot more sense for the pAI to be the modsuit AI.

**Comment by Mothblocks** [comment]
the ai moving you is the fun part that i like though, and that still doesnt solve the problem of that removing the ability to make someone in the existing round an ai and put them into your modsuit

**Comment by Watermelon914** [comment]
there is still a solution here, use MMIs/positronic brains and also make it so that you can choose a custom name when inserted as a positronic brain. That way you can have an identity and still remove AIs from the modsuit equation.

**Comment by davethwave** [comment]
Maybe a tier system of sorts? PAI having less functionality of a real station AI or something I guess but still able to put in a station AI in said modsuits. Perhaps another option that can be used is to remove the ability to put a station AI in and only allow PAI's to be inserted. But to give them more functionality you can make modules that can boost the PAI's ability to interact with the suits. Like a 3 cost module core that allows movement if unconscious or something.

**Comment by TheBonded** [comment]
If we add a tier system roboticists will go back to making AIs for the highest tier possible, which kinda ruins the whole having pAIs as a possibility, if you're going to have something that does it better made in the same place modsuits are made.

**Comment by EricZilla** [comment]
The entire list of why AI in modsuit bad is just a list of reasons you dont like people making pocket AIs and have no actual relation to the modsuit. Every single thing on that list was something that could have been done before modsuits.

**Comment by JohnFulpWillard** [comment]
> The entire list of why AI in modsuit bad is just a list of reasons you dont like people making pocket AIs and have no actual relation to the modsuit. Every single thing on that list was something that could have been done before modsuits.

Building AIs to keep one in your pocket is something I've rarely seen, as intellicarded AIs tend to not be able to do much and is a pretty boring role. Compare this to letting them control your MODsuit, which includes being able to move the player around while not having to worry about subversion like you would while in a shell/mech, and it becomes a way more attractive option to many players including the AI, making it way more common.

**Comment by DrDiasyl** [comment]
"In today episode of stealing from Bay!"
Good PR

File: code/modules/mod/mod_pai.dm [hunk 8]
```diff
@@ -49,7 +49,9 @@
 #define PAI_FALL_TIME 1 SECONDS
 
 /obj/item/mod/control/relaymove(mob/user, direction)
-	if((!active && wearer) || !cell || cell.charge < CELL_PER_STEP  || user != mod_pai || !COOLDOWN_FINISHED(src, cooldown_mod_move) || (wearer?.pulledby?.grab_state))
+	if((!active && wearer) || !cell || cell.charge < CELL_PER_STEP  || user != mod_pai || !COOLDOWN_FINISHED(src, cooldown_mod_move))
+		return FALSE
+	if(wearer && (wearer.pulledby.grab_state || IS_DEAD_OR_INCAP(wearer)))
```
Comment (Fikou) [inline code/modules/mod/mod_pai.dm:54]: IS_DEAD_OR_INCAP is for ai controllers, everything that would make a human have stat would also incapacitate them

**Comment by Fikou** [comment]
> you should keep ais in modsuits and let them move you around but not let pais, that feels like the simplest and best option to me considering i did like the ai-moving interaction quite a bit

i think this is the best solution if the maintainers are disagreeing on what it should do, the movement being on pais and being disabled if anything happens to the user is lame

**Comment by Mothblocks** [comment]
lemon tried to get movement on a doc that collated everyones thoughts, i put mine down

**Comment by LemonInTheDark** [comment]
It went nowhere, will be back to you soon tho

**Comment by Watermelon914** [comment]
>Posibrains aren't assigned names, outside of a list of coder references, until they enter a shell, at which point they will get the name from their preferences menu.
The player cannot choose who takes over the brain nor if the positronic brain even wants to go in a modsuit instead of a borg shell. Compare this to pAI's OOC notes section.
Posibrains are MMIs but intended for ghost roles instead of debraining an actual player. Giving posibrains something MMI's can't would make them more than what they are intended to be.
Like AIs, posibrains are meant to control important roles. AIs, Robots and Mechs are all things that posibrains can be put in, and are all more important roles than sitting in someone's suit.
They don't have laws. What they have to do and who they obey is all up to Policy, something I am not a fan of. This isn't something I can easily change, either, because it would be taking away something that Admins will consider to be part of their domain as Policy, not something that should be forced in code. This is because:
Posibrains aren't Silicons. They're just objects, subtypes of MMIs, actually. There's no silicon involved at all in the brain alone.

**Comment by Mothblocks** [comment]
Closing this PR as it seems we have more or less come to an agreement about the following, though it would be in our best interest to write this down in a design doc. The path we seem to agree on taking is:

- *Replace* the ability to put an AI in a MODsuit with an *MMI*. Let it still do whatever it can do now, with the exception of moving **while dead**. Clarification can be given on that last bit, such as if it should be if the user is stunned/unconscious/whatever, but preventing movement while dead is the *minimum*. Can even potentially be done in a separate PR.
- *Add* the ability to put pAIs in a MODsuit. They cannot move.

**Pull request closed without merging**